src/coreclr/scripts/genEventing.py

```diff
@@ -418,33 +508,69 @@ def generateClrallEvents(eventNodes,allTemplates, target_cpp, runtimeFlavor, wri
             #remove trailing commas
             if len(line) > 0:
                 del line[-1]
+            fnptypeAotNotWindowsSignature = fnptypeAotNotWindowsSignature[:-1]
 
         #add activity IDs
         fnptypeline.append(lindent)
-        fnptypeline.append("%s ActivityId%s\n" % (getEventPipeDataTypeMapping(runtimeFlavor)["LPCGUID"], " = nullptr," if target_cpp else ","))
+        # source file can't have the parameter initializer
+        fnptypeline.append("%s ActivityId%s\n" % (getEventPipeDataTypeMapping(runtimeFlavor)["LPCGUID"], " = nullptr," if (target_cpp and (generatedFileType == "header" or generatedFileType == "header-impl")) else ","))
         fnptypeline.append(lindent)
-        fnptypeline.append("%s RelatedActivityId%s" % (getEventPipeDataTypeMapping(runtimeFlavor)["LPCGUID"], " = nullptr" if target_cpp else ""))
+        fnptypeline.append("%s RelatedActivityId" % (getEventPipeDataTypeMapping(runtimeFlavor)["LPCGUID"]))
+        if  (target_cpp and (generatedFileType == "header" or generatedFileType == "header-impl")):
+            fnptypeline.append(" = nullptr")
+        else:
+            fnptypeline.append("")
 
         fnptype.extend(fnptypeline)
-        fnptype.append("\n)\n{\n")
-        fnbody.append(lindent)
+        fnptype.append("\n)")
+        if generatedFileType == "header" or generatedFileType == "source-impl-noop":
```

Comment:
Redundant given the `if generatedFileType == "header":` / `elif generatedFileType == "source-impl-noop":` below?